core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableMergeJoin.java

```diff
@@ -194,6 +250,81 @@ public static boolean isMergeJoinSupported(JoinRelType joinType) {
     return DeriveMode.BOTH;
   }
 
+  private List<Integer> immutableIntListToList(ImmutableIntList intList, int offset) {
+    ArrayList<Integer> arrayList = new ArrayList<>(intList.size());
+    for (int i : intList) {
+      arrayList.add(i + offset);
+    }
+    return arrayList;
+  }
+
+  private boolean isSubset(List<Integer> a, List<Integer> b) {
+    if (a.size() > b.size()) {
+      return false;
+    }
+    Set<Integer> set = new HashSet<>(b);
+    for (int i = 0; i < a.size(); i++) {
+      if (!set.contains(a.get(i))) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean isPrefixOrderingNotRequired(List<Integer> a, List<Integer> b) {
```

Comment:
See `RelCollations.containsOrderless`, you can make the private one public is necessary.